### PR TITLE
simlify arraycopy

### PR DESCRIPTION
**Pull request by grimreaper**

### CODE REVIEW

File: code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java [hunk 1]
```diff
@@ -210,10 +210,7 @@ public void skillBonusChanged(CharacterLevelEvent e)
 		if (path.length > 0 && StringUtils.isEmpty(String.valueOf(path[path.length - 1])))
 		{
 			displayPath = new Object[path.length - 1];
```
Comment (Shredder121) [inline code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java]: What about?:
```
displayPath = Arrays.copyOf(path, path.length)
```
Just a suggestion, would make it even simpler.

Comment (grimreaper) [inline code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java]: good point: agreed.

Comment (Shredder121) [inline code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java]: And I think I missed this at first, but should it be `path.length - 1`?

File: code/src/java/pcgen/gui2/tools/SpinningTabbedPane.java [hunk 2]
```diff
@@ -314,16 +315,7 @@ private int getMovableTabCount()
...
 
-        int[] list2 = new int[n];
-
-        for (int i = 0; i < n; ++i)
-        {
-            list2[i] = list1[i];
-        }
-
-        //TODO:gorm - test this as a replacement, should be more efficient
-        //System.arraycopy(list1, 1, list2, 1, n - 1);
```
Comment (Shredder121) [inline code/src/java/pcgen/gui2/tools/SpinningTabbedPane.java]: It was supposed to be `n-1`.

File: code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java [hunk 3]
```diff
@@ -209,11 +209,7 @@ public void skillBonusChanged(CharacterLevelEvent e)
 		Object displayPath[];
 		if (path.length > 0 && StringUtils.isEmpty(String.valueOf(path[path.length - 1])))
 		{
-			displayPath = new Object[path.length - 1];
```
Comment (Shredder121) [inline code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java]: This uses `path.length - 1`.

Comment (Zaister) [inline code/src/java/pcgen/gui2/tabs/skill/SkillTreeViewModel.java]: Yup, this breaks the skill tree view TYPE/NAME. I have submitted RP #1910 to fix it.

File: code/src/java/pcgen/gui2/tools/SpinningTabbedPane.java (hunk 2 above)
Comment (Zaister) [inline code/src/java/pcgen/gui2/tools/SpinningTabbedPane.java]: Submitted PR #1912 to fix

**Pull request merged**